### ISSUE TITLE
Consider supporting manual token exchange

### ISSUE THREAD

**Issue by simon-weber** [simon-weber/gpsoauth#44]
As [B16f00t](https://github.com/B16f00t) mentioned in https://github.com/simon-weber/gpsoauth/issues/37#issuecomment-1142402854, this could help in situations where the usual login flow is broken or otherwise impossible. https://github.com/simon-weber/gpsoauth/pull/41 has an example implementation.

**Comment by kiwiz** [comment]
Bump! The manual token exchange appears to be more reliable at the moment. Could support be added for using this flow? I can make updates to the PR as appropiate.

**Comment by simon-weber** [comment]
You're referring to https://github.com/simon-weber/gpsoauth/pull/41? I don't remember much about it. Can you share the code you're using?

**Comment by kiwiz** [comment]
Yup. This [README](https://github.com/rukins/gpsoauth-java#receiving-an-authentication-token) has details on the flow. In short, the endpoint is passed an OAuth token which needs to be manually obtained by a user. The benefit is that it appears to sidestep the authentication issues that users are experiencing. The changes in the PR should still be relevant, but the code could possibly be broken out into a separate method.

**Comment by kiwiz** [comment]
Hi again! Any additional context/investigation work I can provide here?

**Comment by simon-weber** [comment]
I reopened https://github.com/simon-weber/gpsoauth/pull/41.
